src/execution/index/art/iterator.cpp

```diff
@@ -264,14 +270,21 @@ bool Iterator::LowerBound(Node *node, Key &key, bool inclusive) {
 				return Next();
 			}
 		}
+
 		// prefix matches, search inside the child for the key
 		depth += node->prefix.Size();
 
 		top.pos = node->GetChildGreaterEqual(key[depth], equal);
+		// The maximum key byte of the current node is less than the key
+		// So fall back to the previous node
 		if (top.pos == DConstants::INVALID_INDEX) {
```

Comment:
This node will not be leaf, because leaf has been processed above. Maybe you mean to throw exception when node is leaf type. In the previous implementation, `GetChildGreaterEqual` of Node4 will call Node:: GetChildGreaterThan. Maybe I should return `DConstants::INVALID_INDEX` for `Node4, Node16, Node48, Node256` instead of `Node`